### PR TITLE
test: integration test for chained parameters

### PR DESCRIPTION
**Pull request by Bingjiling**
Issue #, if available:

Description of changes:

* Add integration test for chained parameters in search 

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: integration-tests/search.test.ts [hunk 1]
```diff
@@ -53,6 +57,35 @@ describe('search', () => {
         }
     });
 
+    test('search for valid chained parameters', async () => {
+        // The swap is needed so we do not breach the 100 limit in chained parameter results
+        const createParamChainBundle = swapSearchValueToRandomString(cloneDeep(createParameterChainBundle));
```
Comment (rsmayda) [inline integration-tests/search.test.ts]: Should we include a test for this limit?

File: integration-tests/search.test.ts [hunk 2]
```diff
@@ -53,6 +57,35 @@ describe('search', () => {
...
+
+        const response = (await client.post('/', createParamChainBundle)).data;
+        const resources = getResourcesFromBundleResponse(response, createParamChainBundle, true);
+        const testPatient = resources.Patient;
+
+        // wait for the patient to be asynchronously written to ES
+        await waitForResourceToBeSearchable(client, testPatient, true);
```
Comment (rsmayda) [inline integration-tests/search.test.ts]: Why have a partial match here?

File: integration-tests/utils.ts [hunk 3]
```diff
@@ -304,8 +312,57 @@ export const waitForResourceToBeSearchable = async (client: AxiosInstance, resou
...
             },
             resource,
+            partialResourceMatch,
         ),
         20000,
         3000,
     );
 };
+
+export const getResourcesFromBundleResponse = (
```
Comment (rsmayda) [inline integration-tests/utils.ts:425]: Any functional difference in this method?

File: integration-tests/search.test.ts (hunk 2 above)
Comment (Bingjiling) [reply integration-tests/search.test.ts]: Oops, that was a miss. Will remove it and the parameter from the function.

File: integration-tests/utils.ts (hunk 3 above)
Comment (Bingjiling) [reply integration-tests/utils.ts:425]: No, the functionality stays the same.

File: integration-tests/search.test.ts (hunk 1 above)
Comment (Bingjiling) [reply integration-tests/search.test.ts]: Sure, will add it

File: integration-tests/utils.ts [hunk 4]
```diff
@@ -304,8 +312,57 @@ export const waitForResourceToBeSearchable = async (client: AxiosInstance, resou
...
+export const swapSearchValueToRandomString = (
+    originalBundle: any,
+    stringToReplace: string | RegExp = 'randomized-string-for-search',
+) => {
+    const bundleString = JSON.stringify(originalBundle);
+    const regEx = new RegExp(stringToReplace, 'g');
+    const chance = new Chance();
+    const updatedBundleString = bundleString.replace(regEx, chance.word({ length: 15 }));
+    return JSON.parse(updatedBundleString);
+};
```
Comment (carvantes) [inline integration-tests/utils.ts]: IMO this replace is overly complex. It's probably easier to make the `integration-tests/createParameterChainBundle.json` a ts file instead with a function that returns the Bundle object with the random values. Similar to the `randomPatient` method: https://github.com/awslabs/fhir-works-on-aws-deployment/blob/196ecde06442b75676661bc0c27f29c997655bc9/integration-tests/utils.ts#L135

File: integration-tests/search.test.ts [hunk 5]
```diff
@@ -53,6 +55,62 @@ describe('search', () => {
         }
     });
 
+    test('search for valid chained parameters testtest', async () => {
```
Comment (carvantes) [inline integration-tests/search.test.ts]: typo: `testtest`

**Review by rsmayda — DISMISSED** [review]
Looks good! Blocking until package.json is updated with new Search version

**Review by carvantes — APPROVED** [review]

**Review by carvantes — APPROVED** [review]

**Pull request merged**